File: app/src/main/java/co/edu/unipiloto/fuelmanager/normative/NormativePriceActivity.java
package co.edu.unipiloto.fuelmanager.normative;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.button.MaterialButton;

import java.util.List;

import co.edu.unipiloto.fuelmanager.R;
import co.edu.unipiloto.fuelmanager.data.model.NormativePrice;
import co.edu.unipiloto.fuelmanager.normative.NormativePriceRepository;

public class NormativePriceActivity extends AppCompatActivity {

    private MaterialButton          btnActualizar;
    private RecyclerView            recycler;
    private NormativePriceAdapter   adapter;
    private NormativePriceRepository repo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_normative_price);

        repo = new NormativePriceRepository(this);

        btnActualizar = findViewById(R.id.btnActualizarPrecios);
        recycler      = findViewById(R.id.recyclerNormative);

        adapter = new NormativePriceAdapter();
        recycler.setLayoutManager(new LinearLayoutManager(this));
        recycler.setAdapter(adapter);

        // Cargar precios existentes al abrir
        loadPrices();

        btnActualizar.setOnClickListener(v -> actualizarPrecios());
        findViewById(R.id.btnBack).setOnClickListener(v -> finish());
    }

    private void actualizarPrecios() {
        btnActualizar.setEnabled(false);
        btnActualizar.setText("Actualizando...");

        new Thread(() -> {
            boolean ok = repo.fetchAndSaveFromJson();
            List<NormativePrice> lista = repo.getAll();

            runOnUiThread(() -> {
                btnActualizar.setEnabled(true);
                btnActualizar.setText("Actualizar precios normativos");

                if (ok) {
                    adapter.setData(lista);
                    Toast.makeText(this, "Precios actualizados ✓", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(this, "Error al actualizar", Toast.LENGTH_SHORT).show();
                }
            });
        }).start();
    }

    private void loadPrices() {
        new Thread(() -> {
            List<NormativePrice> lista = repo.getAll();
            runOnUiThread(() -> adapter.setData(lista));
        }).start();
    }
}

File: app/src/main/java/co/edu/unipiloto/fuelmanager/Normative/NormativePriceActivity.kt
package co.edu.unipiloto.fuelmanager.normative

import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import co.edu.unipiloto.fuelmanager.Normative.NormativePriceAdapter
import co.edu.unipiloto.fuelmanager.R
import co.edu.unipiloto.fuelmanager.data.repository.NormativePriceRepository

class NormativePriceActivity : AppCompatActivity() {

    private lateinit var btnActualizarPrecios: Button
    private lateinit var normativeRepo: NormativePriceRepository
    private lateinit var recycler: RecyclerView
    private lateinit var adapter: NormativePriceAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_normative_price)

        // Inicializar repository
        normativeRepo = NormativePriceRepository(this)

        // Conectar vistas
        btnActualizarPrecios = findViewById(R.id.btnActualizarPrecios)
        recycler = findViewById(R.id.recyclerNormative)

        // Configurar RecyclerView
        adapter = NormativePriceAdapter()
        recycler.layoutManager = LinearLayoutManager(this)
        recycler.adapter = adapter

        // Listener botón
        btnActualizarPrecios.setOnClickListener {
            actualizarPreciosNormativos()
        }
    }

    private fun actualizarPreciosNormativos() {
        Thread {
            val ok = normativeRepo.fetchAndSaveFromJson()
            val lista = normativeRepo.all

            runOnUiThread {
                if (ok) {
                    Toast.makeText(
                        this,
                        "Precios actualizados ✓",
                        Toast.LENGTH_SHORT
                    ).show()

                    adapter.setData(lista)

                } else {
                    Toast.makeText(
                        this,
                        "Sin conexión",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }
        }.start()
    }
}